testing/kuttl/e2e/replica-read/01-psql-replica-read.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: psql-replica-read
+spec:
+  template:
+    spec:
+      restartPolicy: "OnFailure"
+      containers:
+        - name: psql
+          image: registry.developers.crunchydata.com/crunchydata/crunchy-postgres:centos8-14.1-1
+          command:
+            - "bash"
+            - "-c"
+            # Since we're reading from the replica, pg_is_in_recovery should equal 't'; if it doesn't, exit with an error
+            - "if [`psql -tc 'select pg_is_in_recovery();'` != \"t\" ]; then exit 1; fi"
```

Comment:
Is there a way to check this with SQL so that we can just use the return value? :thinking:

Reply:
PL/pgSQL has [assertions](https://www.postgresql.org/docs/current/plpgsql-errors-and-messages.html#PLPGSQL-STATEMENTS-ASSERT) and is the default language of `DO` statements.

```sql
DO $$ BEGIN ASSERT pg_is_in_recovery(); END $$;
```

🤔 It can be tricky getting strings (like error messages) down that far, so maybe `psql -a` to have it all show up?

```console
$ psql -ac 'DO $$ BEGIN ASSERT pg_is_in_recovery(); END $$;'
DO $$ BEGIN ASSERT pg_is_in_recovery(); END $$;
ERROR:  assertion failed
CONTEXT:  PL/pgSQL function inline_code_block line 1 at ASSERT
```

Reply:
Done